src/main/scala/cromwell/engine/backend/jes/JesAttachedDisk.scala

```diff
@@ -0,0 +1,84 @@
+package cromwell.engine.backend.jes
+
+import com.google.api.services.genomics.model.Disk
+import cromwell.engine.Hashing._
+import cromwell.engine.backend.runtimeattributes.DiskType
+
+import scala.util.Try
+import scalaz.Scalaz._
+import scalaz._
+
+object JesAttachedDisk {
+  val identifier = "[a-zA-Z0-9-_]+"
+  val directory = """/[^\s]+"""
+  val integer = "[0-9]+"
+  val workingDiskPattern = s"""${JesWorkingDisk.Name}\\s+($integer)\\s+($identifier)""".r
+  val mountedDiskPattern = s"""($directory)\\s+($integer)\\s+($identifier)""".r
+
+  def parse(s: String): Try[JesAttachedDisk] = {
+    val validation = s match {
+      case workingDiskPattern(sizeGb, diskType) =>
+        (validateLong(sizeGb) |@| validateDiskType(diskType)) { (s, dt) =>
+          JesWorkingDisk(dt, s.toInt)
+        }
+      case mountedDiskPattern(mountPoint, sizeGb, diskType) =>
+        (validateLong(sizeGb) |@| validateDiskType(diskType)) { (s, dt) =>
+          JesEmptyMountedDisk(dt, s.toInt, mountPoint)
+        }
+      case _ => s"Disk strings should be of the format 'local-disk SIZE TYPE' or '/mount/point SIZE TYPE'".failureNel
+    }
+
+    Try(validation match {
+      case Success(localDisk) => localDisk
+      case Failure(nels) => throw new UnsupportedOperationException(nels.list.mkString("\n"))
+    })
+  }
+
+  private def validateDiskType(diskTypeName: String): ValidationNel[String, DiskType] = {
+    DiskType.values().find(_.diskTypeName == diskTypeName) match {
+      case Some(diskType) => diskType.successNel[String]
+      case None =>
+        val diskTypeNames = DiskType.values.map(_.diskTypeName).mkString(", ")
+        s"Disk TYPE $diskTypeName should be one of $diskTypeNames".failureNel
+    }
+  }
+
+  private def validateLong(value: String): ValidationNel[String, Long] = {
```

Comment:
Not for this PR at all, but we really need to put all those ones in lenthall or somewhere :/